relay-server/src/actors/upstream.rs

```diff
@@ -881,69 +890,115 @@ impl Entry {
 /// is synchronous and managed by the [`UpstreamBroker`].
 #[derive(Debug)]
 struct UpstreamQueue {
+    /// High priority queue.
     high: VecDeque<Entry>,
+    /// Low priority queue.
     low: VecDeque<Entry>,
+    /// Retry queue, sorted by [priority][`RequestPriority`].
+    retries: VecDeque<Entry>,
+    /// Backoff handler for retry entries.
+    retry_backoff: RetryBackoff,
+    /// Retries should not be dequeued before this instant.
+    next_retry: Instant,
```

Comment:
Simplify here. Simply have a single retry delay without backoff, since those retries only run for a _very_ short amount of time before network outage handling kicks in.